log/sequencer.go

```diff
@@ -239,141 +239,140 @@ func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, gu
+		if err != nil {
+			glog.Warningf("%v: Sequencer failed to dequeue leaves: %v", logID, err)
+			return err
+		}
+		seqDequeueLatency.Observe(s.since(stageStart), label)
+		stageStart = s.timeSource.Now()
 
-	// Get the latest known root from storage
-	currentRoot, err := tx.LatestSignedLogRoot(ctx)
-	if err != nil {
-		glog.Warningf("%v: Sequencer failed to get latest root: %v", logID, err)
-		return 0, err
-	}
-	seqGetRootLatency.Observe(s.since(stageStart), label)
-	stageStart = s.timeSource.Now()
-
-	// TODO(al): Have a better detection mechanism for there being no stored root.
-	// TODO(mhs): Might be better to create empty root in provisioning API when it exists
-	if currentRoot.RootHash == nil {
-		glog.Warningf("%v: Fresh log - no previous TreeHeads exist.", logID)
-		// SignRoot starts a new transaction, and we've got one open here until
-		// this function returns.
-		// This explicit Close() is a work-around for the in-memory storage which
-		// locks the tree for each TX.
-		// TODO(al): Producing the first signed root for a new tree should be
-		// handled by the provisioning, move it there.
-		tx.Close()
-		return 0, s.SignRoot(ctx, logID)
-	}
+		// Get the latest known root from storage
+		currentRoot, err := tx.LatestSignedLogRoot(ctx)
+		if err != nil {
+			glog.Warningf("%v: Sequencer failed to get latest root: %v", logID, err)
+			return err
+		}
+		seqGetRootLatency.Observe(s.since(stageStart), label)
+		stageStart = s.timeSource.Now()
+
+		// TODO(al): Have a better detection mechanism for there being no stored root.
+		// TODO(mhs): Might be better to create empty root in provisioning API when it exists
+		if currentRoot.RootHash == nil {
+			glog.Warningf("%v: Fresh log - no previous TreeHeads exist.", logID)
+			// SignRoot starts a new transaction, and we've got one open here until
+			// this function returns.
+			// This explicit Close() is a work-around for the in-memory storage which
+			// locks the tree for each TX.
+			// TODO(al): Producing the first signed root for a new tree should be
+			// handled by the provisioning, move it there.
+			tx.Close()
+			return s.SignRoot(ctx, logID)
+		}
 
-	// There might be no work to be done. But we possibly still need to create an signed root if the
-	// current one is too old. If there's work to be done then we'll be creating a root anyway.
-	numLeaves := len(leaves)
-	if numLeaves == 0 {
-		nowNanos := s.timeSource.Now().UnixNano()
-		interval := time.Duration(nowNanos - currentRoot.TimestampNanos)
-		if maxRootDurationInterval == 0 || interval < maxRootDurationInterval {
-			// We have nothing to integrate into the tree
-			glog.V(1).Infof("%v: No leaves sequenced in this signing operation", logID)
-			return 0, tx.Commit()
+		// There might be no work to be done. But we possibly still need to create an signed root if the
+		// current one is too old. If there's work to be done then we'll be creating a root anyway.
+		numLeaves := len(leaves)
```

Comment:
Should this be `=` not `:=` here so the outer `numLeaves` gets updated?

Reply:
done - good spot, thanks!